lib/include/openamp/rpmsg.h

```diff
@@ -72,7 +73,7 @@ struct rpmsg_endpoint {
 	uint32_t addr;
 	uint32_t dest_addr;
 	rpmsg_ept_cb cb;
-	rpmsg_ept_destroy_cb destroy_cb;
+	rpmsg_ns_unbind_cb unbind_cb;
```

Comment:
I think the destroy callback is used in the destroy endpoint function call, as the destroy endpoint function can be called internally, it will be good to have a callback to notify user the endpoint has been destroyed, the "unbind" endpoint is used for ns destroy name service. I think it can be merged into the rpmsg_ns_bind_cb maybe rename rpmsg_ns_bind_cb to rpmsg_ns_cb()?

Reply:
Please have a look to the commit message, from my POV destroy_cb is not useful.
Furthermore  we need a callback per endpoint for the unbind, not a global call back, has endpoints can be handled by different users.

Reply:
if replace unbind_cb, what should we do with destroy_endpoint()? and who should call destroy_endpoint?
in the destroy_endpoint, we should call unbind_cb() first? and then unregister the endpoint if user want to unregister?
How does user application descides when to remove the endpoint or keep it?
or always assume the unbind_cb() is only called from NS destroy? in this case, should the callback name be more clearer so that user knows it is only called when the NS destroy is received?

Reply:
>if replace unbind_cb, what should we do with destroy_endpoint()? and who should call >destroy_endpoint?
rpmsg_destroy_ept should be called by application when stop the service, to unregister the local end point and send associated NS_DESTROY message.  

>in the destroy_endpoint, we should call unbind_cb() first? and then unregister the endpoint if >user want to unregister?
No, unbind_cb is used only to inform application that remote end point is destroyed. So unbind_cb() can call rpmsg_destroy_ept to desrtroy the local one and free the associated memory ( dynamic allocation case).

>How does user application descides when to remove the endpoint or keep it?
>or always assume the unbind_cb() is only called from NS destroy?
Exactly, unbind_cb() only called on NS_DESTROY to inform that local endpoint is no more bound to the remote one( that is destroyed)

> in this case, should the callback name be more clearer so that user knows it is only called when the NS destroy is received?
What about "ns_unbind_cb" that is symetric with the "ns_bind_cb" defined in  rpmsg_device struct?